agent/otel_metrics.go

```diff
@@ -0,0 +1,55 @@
+package agent
+
+import (
+	"fmt"
+	"math"
+	"time"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+)
+
+type otlpMetrics struct {
+	metrics pmetric.Metrics
+	ms      pmetric.Sum
+}
+
+func newOTLPMetrics() *otlpMetrics {
+	metrics := pmetric.NewMetrics()
+	rm := metrics.ResourceMetrics().AppendEmpty()
+	resourceAttrs := rm.Resource().Attributes()
+	resourceAttrs.PutStr("service.name", "-service")
+	resourceAttrs.PutStr("service.version", "1.0.0")
+	resourceAttrs.PutStr("host.name", "example-host")
+	sm := rm.ScopeMetrics().AppendEmpty()
+	ms := sm.Metrics().AppendEmpty()
+	ms.SetName("bytes_received")
+	sum := ms.SetEmptySum()
+	sum.SetAggregationTemporality(pmetric.AggregationTemporalityDelta)
+	return &otlpMetrics{
+		metrics: metrics,
+		ms:      sum,
+	}
+}
+
+func (om *otlpMetrics) addOTLPSum(timestamp time.Time, value float64, signal usageSignal) error {
+	intVal, err := convertFloat64ToInt64(value)
+	if err != nil {
+		return err
+	}
+	d := om.ms.DataPoints().AppendEmpty()
+	d.SetTimestamp(pcommon.NewTimestampFromTime(timestamp))
+	d.SetIntValue(intVal)
+	d.Attributes().PutStr("signal", string(signal))
+	return nil
+}
+
+func convertFloat64ToInt64(value float64) (int64, error) {
+	if value > math.MaxInt64 {
+		return 0, fmt.Errorf("value %f is too large to convert to int64", value)
+	}
+	if value < math.MinInt64 {
+		return 0, fmt.Errorf("value %f is too small to convert to int64", value)
```

Comment:
"too small" is probably the wrong phrase here -- MinInt64 is a very large negative number. 

I'm also wondering why you bothered to introduce this function at all. It's not actually wrong, but if you had just used int64(value) it would only error if a single report is more than about 9000 terabytes. Then the calling function wouldn't have to return errors either.

Reply:
If there's a overflow issue that causes a datapoint missing, I would like to be able to capture it in refinery's telemetry data. My thinking is that, that's money missing for honeycomb so I want to make sure it's visible

Reply:
OK, fine, it's certainly not wrong. But if someone is overflowing an int64 the money missing would be *at least* in the millions of dollars..  I'd like to have that problem. :)